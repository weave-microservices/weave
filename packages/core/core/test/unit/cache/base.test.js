const { Weave } = require('../../../lib/index')
const { createCacheBase } = require('../../../lib/cache/base')

const config = {
  logger: {
    enabled: false
  }
}

describe('Test base cache factory', () => {
  it('constructor.', () => {
    const broker = Weave(config)
    const baseBroker = createCacheBase(broker)

    expect(baseBroker.log).toBeDefined()
    expect(baseBroker.set).toBeDefined()
    expect(baseBroker.get).toBeDefined()
    expect(baseBroker.remove).toBeDefined()
    expect(baseBroker.clear).toBeDefined()
    expect(baseBroker.middleware).toBeDefined()
    expect(baseBroker.options).toBeDefined()
  })

  it('Options.', () => {
    const broker = Weave(config)
    const baseBroker = createCacheBase(broker)

    expect(baseBroker.options.ttl).toBe(null)
  })

  it('Not implemented methods.', () => {
    const broker = Weave(config)
    const baseBroker = createCacheBase(broker)

    try {
      baseBroker.set('abc', 'def', 5000)
    } catch (error) {
      expect(error.message).toBe('Method not implemented.')
    }

    try {
      baseBroker.get('abc', 'def', 5000)
    } catch (error) {
      expect(error.message).toBe('Method not implemented.')
    }

    try {
      baseBroker.remove()
    } catch (error) {
      expect(error.message).toBe('Method not implemented.')
    }

    try {
      baseBroker.clear()
    } catch (error) {
      expect(error.message).toBe('Method not implemented.')
    }
  })

  it('schould generate a caching hash.', () => {
    const broker = Weave(config)
    const baseBroker = createCacheBase(broker)

    const hash = baseBroker.getCachingHash('test.action', { name: 'Kevin' })
    expect(hash).toMatchSnapshot()
  })

  it('schould generate a caching hash (with 1 key)', () => {
    const broker = Weave(config)
    const baseBroker = createCacheBase(broker)

    const hash = baseBroker.getCachingHash('test.action', { name: 'Kevin', age: 19 }, null, ['name'])
    expect(hash).toMatchSnapshot()
  })

  it('schould generate a caching hash (with 1 key)', () => {
    const broker = Weave(config)
    const baseBroker = createCacheBase(broker)

    const hash = baseBroker.getCachingHash('test.action', { name: 'Kevin', age: 19, hobbies: ['coding', 'gym', 'swimming'], height: null }, null, ['name', 'age', 'hobbies', 'height'])
    expect(hash).toMatchSnapshot()
  })
})

