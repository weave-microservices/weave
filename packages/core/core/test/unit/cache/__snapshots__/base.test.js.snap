// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test base cache factory schould generate a caching hash (with 1 key) 1`] = `"test.action:Kevin"`;

exports[`Test base cache factory schould generate a caching hash (with 1 key) 2`] = `"test.action:Kevin|19|vpXyfjaPuF2ldvswluY4C+gHCdI=|null"`;

exports[`Test base cache factory schould generate a caching hash. 1`] = `"test.action:zo2xLEbtdnteSnpO9Gj7KD1+wqw="`;
